src/ngx_http_vhost_traffic_status_string.c

```diff
@@ -187,7 +187,7 @@ ngx_http_vhost_traffic_status_escape_prometheus(ngx_pool_t *pool, ngx_str_t *buf
             }
         } else {
             char_end = pa;
-            if (ngx_utf8_decode(&char_end, last - pa) > 0x10ffff) {
+            if (*char_end > 0xf8 || ngx_utf8_decode(&char_end, last - pa) > 0x10ffff) {
```

Comment:
```suggestion
            if (*pa >= 0xf8 || ngx_utf8_decode(&char_end, last - pa) > 0x10ffff) {
```
This is a good approach, but `0xf8` is invalid too, so you need to use `>=`.

I also suggest using `*pa` as this is the pointer to the current character. `char_end` happens to have the same value, but it has a different meaning associated with its use in the `ngx_utf8_decode` function.

Reply:
@jongiddy Thanks! I fixed it.